duckduckgo/iOS
# Subscriptions. 28. Intercept Privacy Pro URL + Navigation Update Ship review fixes

Task/Issue URL:  https://app.asana.com/0/0/1206983541370659/f
Tech Design URL:
CC:

**Description**:
- Implements a URLInterceptor class to define custom actions when visiting specific URLs
- Other UI fixes based on Ship Review for navigation Updates
- Minor updates to cancellable removals and deInit cleanUp



**Steps to test this PR**:
1.  Go to any website
2. Go to https://sbarag.duckduckgo.com/pro 
3. Go back until you get to the browser, you should be on the first page you loaded
4. Open a new tab and go to https://codepen.io/marcosholgado/pen/PogEJrM
5. Click on Click Here
6. You should be taken to the native flow
7. Go back to the browser, you should be in the same page as before
8. Long click on click here and open in a new tab
9. You should be taken to the native flow
10. Go back to the browser, you should be in the same page
11. Long click on click here and open in a new background tab
12. You should be taken to the native flow
